mod/primitives/pkg/encoding/ssz/schema/schema.go

```diff
@@ -0,0 +1,205 @@
+func UInt32() SSZType  { return basic{size: 4} }
+func UInt64() SSZType  { return basic{size: 8} }
+func UInt128() SSZType { return basic{size: 16} }
+func UInt256() SSZType { return basic{size: 32} }
+
+func (b basic) Size() uint64 { return b.size }
+
+func (b basic) Chunks() uint64 { return 1 }
+
+func (b basic) child(_ string) SSZType { return b }
+
+func (b basic) position(_ string) (uint64, uint8, error) {
+	return 0, 0, errors.New("basic type has no children")
+}
+
+// Container Type
+
+type container struct {
+	Fields     []SSZType
+	FieldIndex map[string]uint64
+}
+
+type field struct {
+	name string
+	typ  SSZType
+}
+
+func Field(name string, typ SSZType) field {
+	return field{name: name, typ: typ}
+}
+
+func Container(fields ...field) SSZType {
+	fieldIndex := make(map[string]uint64)
+	types := make([]SSZType, len(fields))
+	for i, f := range fields {
+		fieldIndex[f.name] = uint64(i)
+		types[i] = f.typ
+	}
+	return container{Fields: types, FieldIndex: fieldIndex}
+}
+
+func (c container) Size() uint64 { return chunkSize }
+
+func (c container) Length() uint64 { return uint64(len(c.Fields)) }
+
+func (c container) Chunks() uint64 { return uint64(len(c.Fields)) }
+
+func (c container) child(p string) SSZType {
+	return c.Fields[c.FieldIndex[p]]
+}
+
+func (c container) position(p string) (uint64, uint8, error) {
+	pos, ok := c.FieldIndex[p]
+	if !ok {
+		return 0, 0, fmt.Errorf("field %s not found", p)
+	}
+	return pos, 0, nil
+}
+
+// Enumerable Type (vectors and lists)
+
+func List(element SSZType, length uint64) SSZType {
+	return enumerable{Element: element, maxLength: length}
+}
+
+func Vector(element SSZType, length uint64) SSZType {
+	return enumerable{Element: element, length: length}
+}
+
+func Bytes(length uint64) SSZType {
+	return Vector(UInt8(), length)
+}
+
+type enumerable struct {
+	Element   SSZType
+	length    uint64
+	maxLength uint64
+}
+
+func (e enumerable) Size() uint64 { return chunkSize }
+
+func (e enumerable) Chunks() uint64 {
+	x := float64(e.Length()*e.Element.Size()) / chunkSize
+	return uint64(math.Ceil(x))
+}
+
+func (e enumerable) child(_ string) SSZType {
+	return e.Element
+}
+
+func (e enumerable) Length() uint64 {
+	if e.length == 0 {
+		return e.maxLength
+	}
+	return e.length
+}
+
+func (e enumerable) position(p string) (uint64, uint8, error) {
+	i, err := strconv.ParseUint(p, 10, 64)
+	if err != nil {
+		return 0, 0, fmt.Errorf("expected index, got name %s", p)
+	}
+	start := i * e.Element.Size()
+	return uint64(math.Floor(float64(start) / chunkSize)),
+		uint8(start % chunkSize),
+		nil
+}
+
+func (e enumerable) IsByteVector() bool {
+	return e.Element.Size() == 1 && e.length > 0
+}
+
+func (e enumerable) IsList() bool {
+	return e.maxLength > 0
+}
+
+func (e enumerable) IsFixed() bool {
+	_, ok := e.Element.(basic)
+	return ok
+}
+
+type Node struct {
+	SSZType
+
+	GIndex uint64
+	Offset uint8
+}
+
+func Path(path string) []string {
+	return strings.Split(path, "/")
```

Comment:
What was your rationale again for not doing []int everywhere?

Reply:
we could create a hard type 

ObjectPath []int

Reply:
I like how the spec described object paths this way, they're human readable and writable.  I think its easier for a dev to write and maintain e.g. `"validator/n"`vs `10/n` when e.g. [retrieving a validator by index](https://github.com/berachain/beacon-kit/blob/d69a1941513c852e7b10ff4d67db6cfd561ad772/mod/storage/pkg/beacondb/registry.go#L128). 

Also the length node for lists should be directly addressable somehow, right now the token `"__len__"` is used, for e.g. [GetTotalValidators](https://github.com/berachain/beacon-kit/blob/d69a1941513c852e7b10ff4d67db6cfd561ad772/mod/storage/pkg/beacondb/registry.go#L169).

Either way can work though.

Reply:
I'm good with following spec!